packages/design-system/src/components/VerticalNav/VerticalNavItem.tsx

```diff
@@ -1,137 +1,195 @@
+  const id = props.id || uniqueId('VerticalNavItem_');
+  const subnavId = `${id}__subnav`;
 
-    return this.handleClick(evt);
-  }
+  const [collapsed, setCollapsed] = useState(props.defaultCollapsed);
+
+  // Refactor from `componentDidUpdate` to `isEffect` using this example:
+  // https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
+  React.useEffect(() => {
+    if (props.onSubnavToggle) {
+      props.onSubnavToggle(props.id, collapsed);
+    }
+  }, [collapsed]);
 
   /**
    * Note: This event handler will only get called when the VerticalNavItemLabel
    * is a link or plain text
    */
-  handleClick(evt) {
-    if (this.props.onClick) {
-      this.props.onClick(evt, this.id, this.props.url);
+  const handleClick = (evt: React.MouseEvent | React.KeyboardEvent): void => {
+    if (props.onClick) {
+      props.onClick(evt, id, props.url);
     }
-  }
+  };
 
-  handleToggleClick() {
-    this.setState({ collapsed: !this.state.collapsed });
-  }
+  const handleToggleClick = (): void => {
+    setCollapsed(!collapsed);
+  };
 
-  hasSubnav() {
-    return Boolean(this.props.items && this.props.items.length > 0);
-  }
+  const hasSubnav = (): boolean => Boolean(props.items && props.items.length > 0);
 
   /**
-   * Check if this item is selected or if it is a parent of a selected item
-   * @return {Boolean}
+   * Called when VerticalNavItemLabel is clicked. Since the "label" could be
+   * a link, subnav toggle button, or plain text, we use this method to
+   * determine what action to take and which event to actually fire.
+   * @param {Object} SyntheticEvent
    */
-  isSelected() {
-    if (this.props.selected) return this.props.selected;
-
-    if (this.props._selectedId && this.hasSubnav()) {
-      return this.childIsSelected(this.props.items);
+  const handleLabelClick = (evt: React.MouseEvent | React.KeyboardEvent): void => {
+    if (hasSubnav()) {
+      return handleToggleClick();
     }
 
-    return false;
-  }
+    return handleClick(evt);
+  };
 
   /**
    * Checks if a descendant is selected
    * @param {Array} children - The nested items
    * @return {Boolean}
    */
-  childIsSelected(children) {
+  const childIsSelected = (children): boolean => {
     if (children && children.length) {
-      return children.some((child) => {
-        return child.id === this.props._selectedId || this.childIsSelected(child.items);
+      return children.some((child: any) => {
+        return child.id === props._selectedId || childIsSelected(child.items);
       });
     }
 
     return false;
-  }
+  };
+
+  /**
+   * Check if this item is selected or if it is a parent of a selected item
+   * @return {Boolean}
+   */
+  const isSelected = (): boolean => {
+    if (props.selected) return props.selected;
+
+    if (props._selectedId && hasSubnav()) {
+      return childIsSelected(props.items);
+    }
+
+    return false;
+  };
```

Comment:
It looks like this is only called in one place, so it could be a plain value instead of a function